service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -0,0 +1,154 @@
+
+    private EhrExtractMapper ehrExtractMapper;
+    private MessageContext messageContext;
+
+    @BeforeEach
+    public void setUp() {
+        getGpcStructuredTaskDefinition = GetGpcStructuredTaskDefinition.builder()
+            .nhsNumber(TEST_NHS_NUMBER)
+            .conversationId(TEST_CONVERSATION_ID)
+            .requestId(TEST_REQUEST_ID)
+            .fromOdsCode(TEST_FROM_ODS_CODE)
+            .toOdsCode(TEST_TO_ODS_CODE)
+            .build();
+
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID_1, TEST_ID_2, TEST_ID_3);
+        when(timestampService.now()).thenReturn(Instant.parse(TEST_DATE_TIME));
+        when(codeableConceptCdMapper.mapCodeableConceptToCd(any(CodeableConcept.class)))
+            .thenReturn(CodeableConceptMapperMockUtil.NULL_FLAVOR_CODE);
+        messageContext = new MessageContext(randomIdGeneratorService);
+        EncounterComponentsMapper encounterComponentsMapper = new EncounterComponentsMapper(
+            messageContext,
+            new DiaryPlanStatementMapper(messageContext, codeableConceptCdMapper),
+            new NarrativeStatementMapper(messageContext),
+            new ObservationStatementMapper(
+                messageContext,
+                new StructuredObservationValueMapper(),
+                new PertinentInformationObservationValueMapper(),
+                codeableConceptCdMapper
+            ),
+            new ImmunizationObservationStatementMapper(messageContext, codeableConceptCdMapper),
+            new ConditionLinkSetMapper(messageContext, randomIdGeneratorService, codeableConceptCdMapper),
+            new BloodPressureMapper(
+                messageContext, randomIdGeneratorService, new StructuredObservationValueMapper(), codeableConceptCdMapper)
+        );
+
+        ehrExtractMapper = new EhrExtractMapper(randomIdGeneratorService,
+            timestampService,
+            new EncounterMapper(messageContext, encounterComponentsMapper));
+    }
+
+    @AfterEach
+    public void tearDown() {
+        messageContext.resetMessageContext();
+    }
+
+    @ParameterizedTest
+    @MethodSource("testValueFilePaths")
+    public void When_MappingProperJsonRequestBody_Expect_ProperXmlOutput(String input) throws IOException {
+        String inputJsonFileContent = ResourceTestFileUtils.getFileContent(TEST_FILES_DIRECTORY + input);
+        Bundle bundle = new FhirParseService().parseResource(inputJsonFileContent, Bundle.class);
+        messageContext.initialize(bundle);
+
+        EhrExtractTemplateParameters ehrExtractTemplateParameters = ehrExtractMapper.mapBundleToEhrFhirExtractParams(
+            getGpcStructuredTaskDefinition,
+            bundle);
+        String output = ehrExtractMapper.mapEhrExtractToXml(ehrExtractTemplateParameters);
+
+        //TODO: compare with the proper XML files
+        assertThat(output).isEqualTo(output);
+    }
+
+    private static Stream<Arguments> testValueFilePaths() {
```

Comment:
We can use `List.of()` here, then we don't have to wrap each string in `Arguments.of()`

Reply:
I actually just have updated the structure of EhrExtractClass a bit and uploaded some stub XML files for each json we have. So, now the Arguments.of() takes 2 parameters instead of one. The input and output file. Do you think it can stay as it is in this case?

Reply:
Ah ok, no problem 👍